File: sdks/go/pkg/models/operations/postfoldersidmetadataidid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostFoldersIDMetadataIDIDScopeEnum string

const (
	PostFoldersIDMetadataIDIDScopeEnumGlobal     PostFoldersIDMetadataIDIDScopeEnum = "global"
	PostFoldersIDMetadataIDIDScopeEnumEnterprise PostFoldersIDMetadataIDIDScopeEnum = "enterprise"
)

type PostFoldersIDMetadataIDIDPathParams struct {
	FolderID    string                             `pathParam:"style=simple,explode=false,name=folder_id"`
	Scope       PostFoldersIDMetadataIDIDScopeEnum `pathParam:"style=simple,explode=false,name=scope"`
	TemplateKey string                             `pathParam:"style=simple,explode=false,name=template_key"`
}

type PostFoldersIDMetadataIDIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams PostFoldersIDMetadataIDIDPathParams
	Request    map[string]string `request:"mediaType=application/json"`
}

type PostFoldersIDMetadataIDIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	Metadata    *shared.Metadata
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/postfoldersidmetadataidid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum PostFoldersIdMetadataIdIdScopeEnum {
    Global = "global",
    Enterprise = "enterprise"
}


export class PostFoldersIdMetadataIdIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=folder_id" })
  folderId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=scope" })
  scope: PostFoldersIdMetadataIdIdScopeEnum;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=template_key" })
  templateKey: string;
}


export class PostFoldersIdMetadataIdIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PostFoldersIdMetadataIdIdPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: Record<string, string>;
}


export class PostFoldersIdMetadataIdIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  metadata?: shared.Metadata;

  @SpeakeasyMetadata()
  statusCode: number;
}
